### DISCUSSION TITLE
Reusing nested pipelines

### DISCUSSION THREAD

**Discussion by roytman** [kubeflow/pipelines#10970] [General]
KFPv2 introduced nested pipelines. Would it be possible to reuse a nested pipeline definition as we can with shared components?

Let me explain my scenario. In our project, we use Ray, so we have a number of simple pipelines that usually contain components to start a Ray cluster, execute a job, and remove the Ray cluster. These pipelines can also include other components/tasks.

Now, I would like to combine several of these simple pipelines into a single "super" pipeline. While I could create one large pipeline definition file with all the nested pipelines, the file would be huge, and I would prefer to reuse the existing simple pipeline definitions.

It would be ideal if we could create a nested pipeline from files (Python or YAML).
What do you think?

**Comment by chensun** [comment]
Pipelines and components share the same YAML format, you can reuse and nest pipelines just like what you would do with components. 
https://www.kubeflow.org/docs/components/pipelines/user-guides/components/load-and-share-components/

**Reply by roytman** [reply]
Thank you @chensun
